media-sound/pulseaudio/pulseaudio-14.2.ebuild

```diff
@@ -0,0 +1,318 @@
+	jack? ( virtual/jack )
+	tcpd? ( sys-apps/tcp-wrappers[${MULTILIB_USEDEP}] )
+	lirc? ( app-misc/lirc )
+	dbus? ( >=sys-apps/dbus-1.0.0[${MULTILIB_USEDEP}] )
+	gtk? ( x11-libs/gtk+:3 )
+	bluetooth? (
+		>=net-wireless/bluez-5
+		media-libs/sbc
+	)
+	asyncns? ( net-libs/libasyncns[${MULTILIB_USEDEP}] )
+	udev? ( >=virtual/udev-143[hwdb(+)] )
+	equalizer? (
+		sci-libs/fftw:3.0
+	)
+	ofono-headset? ( >=net-misc/ofono-1.13 )
+	orc? ( >=dev-lang/orc-0.4.15 )
+	sox? ( >=media-libs/soxr-0.1.1 )
+	ssl? ( dev-libs/openssl:0= )
+	media-libs/speexdsp
+	gdbm? ( sys-libs/gdbm:= )
+	webrtc-aec? ( >=media-libs/webrtc-audio-processing-0.2 )
+	systemd? ( sys-apps/systemd:0=[${MULTILIB_USEDEP}] )
+	!systemd? ( sys-auth/elogind )
+	dev-libs/libltdl:0
+	selinux? ( sec-policy/selinux-pulseaudio )
+" # libltdl is a valid RDEPEND, libltdl.so is used for native abi in pulsecore and daemon
+
+DEPEND="${RDEPEND}
+	X? (
+		x11-base/xorg-proto
+		>=x11-libs/libXtst-1.0.99.2[${MULTILIB_USEDEP}]
+	)
+	dev-libs/libatomic_ops
+"
+# This is a PDEPEND to avoid a circular dep
+PDEPEND="
+	alsa? ( alsa-plugin? ( >=media-plugins/alsa-plugins-1.0.27-r1[pulseaudio,${MULTILIB_USEDEP}] ) )
+"
+
+# alsa-utils dep is for the alsasound init.d script (see bug #155707); TODO: read it
+RDEPEND="${RDEPEND}
+	system-wide? (
+		alsa? ( media-sound/alsa-utils )
+		acct-user/pulse
+		acct-group/pulse-access
+	)
+	acct-group/audio
+"
+
+BDEPEND="
+	doc? ( app-doc/doxygen )
+	system-wide? ( dev-util/unifdef )
+	test? ( >=dev-libs/check-0.9.10 )
+	sys-devel/gettext
+	sys-devel/m4
+	virtual/pkgconfig
+"
+
+# BUG: work-around for a weird bug
+#DOCS=( NEWS README ) # todo is useless to install
+
+#PATCHES=()
+
+pkg_pretend() {
+	CONFIG_CHECK="~HIGH_RES_TIMERS"
+	WARNING_HIGH_RES_TIMERS="CONFIG_HIGH_RES_TIMERS:\tis not set (required for enabling timer-based scheduling in pulseaudio)\n"
+	check_extra_config
+
+	if linux_config_exists; then
+		local snd_hda_prealloc_size=$(linux_chkconfig_string SND_HDA_PREALLOC_SIZE)
+		if [ -n "${snd_hda_prealloc_size}" ] && [ "${snd_hda_prealloc_size}" -lt 2048 ]; then
+			ewarn "A pre-allocated buffer-size of 2048 (kB) or higher is recommended for the"
+			ewarn "HD-audio driver when using timer-based scheduling!"
+			ewarn "CONFIG_SND_HDA_PREALLOC_SIZE=${snd_hda_prealloc_size}"
+		fi
+	fi
+}
+
+pkg_setup() {
+	linux-info_pkg_setup
+	gnome2_environment_reset #543364
+}
+
+#src_prepare() {
+#	default
+
+	# Skip test that cannot work with sandbox, bug #501846
+	#sed -i -e '/lock-autospawn-test /d' src/Makefile.am || die
+	#sed -i -e 's/lock-autospawn-test$(EXEEXT) //' src/Makefile.in || die
+#}
+
+pa_meson_multilib_native_use_enable() {
+	echo "-D ${2:-${1}}=$(multilib_native_usex ${1} true false)"
+}
+
+pa_meson_multilib_native_use_feature() {
+	echo "-D ${2:-${1}}=$(multilib_native_usex ${1} enabled disabled)"
+}
+
+multilib_src_configure() {
+	local emesonargs=(
+		-Dadrian-aec=false # Not packaged?
+		--localstatedir="${EPREFIX}"/var
```

Comment:
Yes, it does stand out but, if my memory serves me right, when disabled, it put the runtime files under /var/lib/tmp or some other similarly obscure path that's unlikely to get automatically cleaned out. I will make a TODO item to recheck this at some point.

Reply:
FWIW, we have to do this in `gnome-base/gdm` as well. Their build system seems to assume `localstatedir` should be `/var` rather than `/var/lib`, so this isn't too surprising to me.

Reply:
Ah, this is where my lack of GNOME-fu matters :)

Reply:
On retest this is what I see:
    localstatedir:                 /var/lib
...
    System Runtime Path:           /var/lib/run/pulse
    System State Path:             /var/lib/lib/pulse
    System Config Path:            /var/lib/lib/pulse

Most obviously the /var/lib/lib looks immediately weird. Setting --localstatedir=/var removes that lib for more normal looking paths but no idea if they are correct according to any standard that might apply. ;)